content/03.methods.md

```diff
@@ -87,6 +87,50 @@ $$\hat{\theta} = argmin_{\theta} -\sum_{\Lambda} log \sum_{Y}P(\Lambda, Y)$$
 
 In the framework we used predictions from the generative model, $\hat{Y} = P(Y \mid \Lambda)$, as training classes for our dataset [@doi:10.14778/3157794.3157797; @doi:10.1145/3209889.3209898]. 
 
+### Discriminative Model
+
+The discriminative model is a neural network, which we trained using labels predicted from the generative model.
+The expectation is that the discriminative model can improve upon the generative model's predictions by incorporating textual features along with generated training labels.
+We used a piecewise convolutional neural network that contains multiple kernel filters as our discriminative model.
+This network uses multiple filters with fixed widths of 300 dimensions and a fixed height of 7 (Figure {@fig:convolutional_network}).
+We choose a fixed height of 7 because this height was previously established to provide the best performance in relationship classification [@arxiv:1510.03820].
+We trained this model for 15 epochs using the Adam optimizer [@arxiv:1412.6980] with pytorch's default parameter settings and a learning rate of 0.001 that decreases by half every epoch until the lower bound of 1e-5 is reached.
```

Comment:
```suggestion
We trained this model for 15 epochs using the Adam optimizer [@arxiv:1412.6980] with pytorch's default parameter settings and a learning rate of 0.001 that decreases by half every epoch until the lower bound of 1e-5 is reached, which we observed was often sufficient for convergence.
```
I think you showed me results that suggested that 15 epochs was sufficient by looking at convergence. Check to be sure. This is a bit of results-like text, but I think the point is unrelated enough to your main paper that it could go in here & it helps the reader understand why you stopped at 15.

Reply:
Sending a note to confirm that 15 epochs is sufficient for convergence.